src/EasyWebWorker.ts

```diff
@@ -472,22 +476,27 @@ export class EasyWebWorker<TPayload = null, TResult = void> {
    * @returns {IMessagePromise<TResult>} generated defer that will be resolved when the message completed
    */
   public send = ((
-    ...payload: TPayload extends null ? [null?] : [TPayload]
+    payload: TPayload,
+    transfer?: Transferable[]
   ): CancelablePromise<TResult> => {
-    const [$payload] = payload as [TPayload];
-
-    return this.sendToWorker<TPayload, TResult>({ payload: $payload });
+    return this.sendToWorker<TPayload, TResult>({ payload }, transfer);
   }) as unknown as TPayload extends null
     ? () => CancelablePromise<TResult>
-    : (payload: TPayload) => CancelablePromise<TResult>;
-
-  private sendToWorker = <TPayload_ = null, TResult_ = void>({
-    payload,
-    method,
-  }: {
-    payload?: TPayload_;
-    method?: string;
-  }): CancelablePromise<TResult_> => {
+    : (
+        payload: TPayload,
+        transfer?: Transferable[]
+      ) => CancelablePromise<TResult>;
```

Comment:
The type annotations you're using here can cause odd typing errors. For instance, if you have a union type as your `TPayload`, like this:

```ts
type MyPayload = {
  type: "first",
  foo: number,
} | {
  type: "second",
  bar: string
};
```

Then, when you use the `send` function, TypeScript expects the `payload` argument to be `never`, which doesn't make sense. This could be because TypeScript treats a union type as `extends null`. I found that removing the type annotation from my version solves this problem. So, you might want to consider doing that, although I'm not completely sure why it was added in the first place.

Reply:
Please check my related comment here: https://github.com/johnny-quesada-developer/easy-web-worker/issues/6#issue-2074487212

Reply:
ChatGPT suggests:

> Use Function Overloading: TypeScript supports function overloading, which allows you to define multiple function signatures for the same function. You could define multiple versions of the send and resolve functions, each with a different signature that explicitly handles different types or union types.